### PR TITLE
Update kubetest azure to use ACR

### PR DESCRIPTION
**Pull request by ritazh**
- Add support for ACR in addition to Docker
- Update job config for pull-kubernetes-e2e-aks-engine-azure
- Bump version of aks-engine to support `privateAzureRegistryServer`

NOTE: A new `kubekins-e2e` will need to be created after this PR is merged. Thank you!!!

cc @feiskyer @justaugustus @adelina-t @BenTheElder

### CODE REVIEW

File: config/jobs/kubernetes-security/generated-security-jobs.yaml [hunk 1]
```diff
@@ -434,11 +434,11 @@ presubmits:
...
         - --timeout=420m
         - --stage=gs://kubernetes-security-prow/ci/pull-security-kubernetes-e2e-aks-engine-azure
-        image: gcr.io/k8s-testimages/kubekins-e2e:v20190219-caa48e6c3-master
+        image: ritazh/kubekins-e2e:v20190220-d69e2d7-dirty-master
```
Comment (feiskyer) [inline config/jobs/kubernetes-security/generated-security-jobs.yaml]: should we use some tag build from `gcr.io/k8s-testimages`?

Comment (ritazh) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: Sorry forgot to change this back. Just restored to the old image, but will need to create another PR once we get a new kubekins-e2e image after this PR is merged.

**Comment by andyzhangx** [comment]
@wojtek-t @krzyzacy @bentheelder could you take a look.
one k8s upstream e2e test is depending on this test config, thanks.

**Comment by ritazh** [comment]
@BenTheElder can you pls help build a new image of kubekins-e2e?

**Comment by BenTheElder** [comment]
It's up now, had to update the tools on my laptop first: https://github.com/kubernetes/test-infra/pull/11381

**Pull request merged**